src/mobile/scripts/Carousel.js

```diff
@@ -0,0 +1,295 @@
+(function (window, ch, Carousel) {
+    'use strict';
+
+    /**
+     * rAF request Animation Frame
+     * @private
+     * @type Function()
+     */
+    var rAF = (function(){
+        return  window.requestAnimationFrame ||
+                window.webkitRequestAnimationFrame ||
+                window.mozRequestAnimationFrame ||
+                function (callback) {
+                    window.setTimeout(callback, 1000 / 60);
+                };
+    }());
+
+    Carousel.prototype.addTouchEvents = function () {
+
+        /**
+         * Reference to context of an instance.
+         * @type {Object}
+         * @private
+         */
+        var that = this;
+            
+        /**
+         * Element that wraps the list and denies its overflow.
+         * @private
+         */
+        this.mask = this._$mask[0];
+
+        /**
+         * Element that moves (slides) across the component (inside the mask).
+         * @private
+         */
+        this.list = this._$list[0];
+
+        /**
+         * The X position of the first touch.
+         * @private
+         * @type {Number}
+         */
+        this._startTouchX = 0;
+
+        /**
+         * The Y position of the first touch.
+         * @private
+         * @type {Number}
+         */
+        this._startTouchY = 0;
+         
+        /**
+         * The X position of the final touch.
+         * @private
+         * @type {Number}
+         */
+        this._endTouchX = 0;
+
+        /**
+         * The X position of the last touch.
+         * @private
+         * @type {Number}
+         */
+        this._partialTouchX;
```

Comment:
Is this OK? Don't we need to do an assignment here like: `this._partialTouchX = 0;`?